libcloudforensics/providers/gcp/internal/storagetransfer.py

```diff
@@ -0,0 +1,163 @@
+
+logging_utils.SetUpLogger(__name__)
+logger = logging_utils.GetLogger(__name__)
+
+if TYPE_CHECKING:
+  import googleapiclient
+
+
+class GoogleCloudStorageTransfer:
+  """Class to call Google Cloud Storage Transfer APIs.
+
+  Attributes:
+    gcst_api_client: Client to interact with GCST APIs.
+    project_id: Google Cloud project ID.
+  """
+  CLOUD_STORAGE_TRANSFER_API_VERSION = 'v1'
+
+  def __init__(self, project_id: Optional[str] = None) -> None:
+    """Initialize the GoogleCloudStorageTransfer object.
+
+    Args:
+      project_id (str): Optional. Google Cloud project ID.
+    """
+
+    self.gcst_api_client = None
+    self.project_id = project_id
+
+  def GcstApi(self) -> 'googleapiclient.discovery.Resource':
+    """Get a Google Cloud Storage Transfer service object.
+
+    Returns:
+      googleapiclient.discovery.Resource: A Google Cloud Storage Transfer
+        service object.
+    """
+
+    if self.gcst_api_client:
+      return self.gcst_api_client
+    self.gcst_api_client = common.CreateService(
+        'storagetransfer', self.CLOUD_STORAGE_TRANSFER_API_VERSION)
+    return self.gcst_api_client
+
+  def S3ToGCS(self, s3_path: str, zone: str, gcs_path: str) -> Dict[str, Any]:
+    """Copy an S3 object to a GCS bucket.
+
+    Args:
+      s3_path (str): File path to the S3 resource.
+          Ex: s3://test/bucket/obj
+      zone (str): The AWS zone in which resources are located.
+        Available zones are listed at:
+        https://cloud.google.com/storage-transfer/docs/create-manage-transfer-program#s3-to-cloud  # pylint: disable=line-too-long
+      gcs_path (str): File path to the target GCS bucket.
+          Ex: gs://bucket/folder
+    Returns:
+      Dict: An API operation object for a Google Cloud Storage Transfer operation.
+        https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferOperations/list  # pylint: disable=line-too-long
+    Raises:
+      TransferCreationError: If the transfer couldn't be created.
+      TransferExecutionError: If the transfer couldn't be run.
+    """
+    aws_creds = account.AWSAccount(zone).session.get_credentials()
+    s3_bucket, s3_path = SplitStoragePath(s3_path)
+    gcs_bucket, gcs_path = SplitStoragePath(gcs_path)
+    if not gcs_path.endswith('/'):
+      gcs_path = gcs_path + '/'
+    # Don't specify a path if we're writing to the bucket root.
+    if gcs_path == '/':
+      gcs_path = ''
+    today = datetime.datetime.now()
+    transfer_job_body = {
+        'projectId': self.project_id,
+        'description': 'created_by_cfu',
+        'transferSpec': {
+            'objectConditions': {
+                'includePrefixes': [s3_path]
+            },
+            'awsS3DataSource': {
+                'bucketName': s3_bucket,
+                'awsAccessKey': {
+                    'accessKeyId': aws_creds.access_key,
+                    'secretAccessKey': aws_creds.secret_key
```

Comment:
How does this behave if the user is authenticated to AWS via a temporary session? Should we conditionally set a session token as well?

Reply:
As discussed offline, bailing if no long term creds